internal/tuf/tuf.go

```diff
@@ -338,3 +357,22 @@ type OtherRepository interface {
 	// sign the other repository's initial gittuf root of trust metadata.
 	GetInitialRootPrincipals() []Principal
 }
+
+// Applet represents a hook entry in a rule file ('TargetsMetadata').
+type Applet interface {
```

Comment:
I feel like we should at least align the interface methods (AddHook) with this interface...